lib/src/git.rs

```diff
@@ -68,11 +68,32 @@ pub fn import_refs(
     git_repo: &git2::Repository,
     git_settings: &GitSettings,
 ) -> Result<(), GitImportError> {
+    import_some_refs(mut_repo, git_repo, git_settings, |_| true)
+}
+
+/// Reflect changes made in the underlying Git repo in the Jujutsu repo.
+/// If `git_ref_filter` is defined, only branches whose git full reference name
+/// pass the filter will be considered for addition, update, or deletion.
+pub fn import_some_refs<F>(
+    mut_repo: &mut MutableRepo,
+    git_repo: &git2::Repository,
+    git_settings: &GitSettings,
+    git_ref_filter: F,
+) -> Result<(), GitImportError>
+where
+    F: Fn(&str) -> bool,
+{
     let store = mut_repo.store().clone();
     let mut existing_git_refs = mut_repo.view().git_refs().clone();
     let mut old_git_heads = existing_git_refs
-        .values()
-        .flat_map(|old_target| old_target.adds())
+        .iter()
+        .flat_map(|(ref_name, old_target)| {
+            if git_ref_filter(ref_name) {
+                old_target.adds()
+            } else {
+                vec![]
+            }
+        })
```

Comment:
I feel like we should not make this change. Let's say the repo looked like this before:

```
C branch2
|
B branch1
|
A
```

If  branch1 is deleted in git, then the next full import would not abandon B because branch2 still points to it. If we do a partial import of only branch1, however, it looks like we'd then abandon commit B, right? Could you add a test case for this scenario?

Maybe the fix is to add old heads that don't match the filter to `new_git_heads`.

Reply:
Indeed. In fact, the problem will happen if branch1 and branch2 are both deleted in git and we only import branch1. In this case, we will abandon B while branch2, although deleted, should be left untouched.

As you suggested, I've added non-matching heads to `new_git_heads` in order to preserve them, and added a test which fails when we do not do this.

I have also added a similar test in the CLI.